core/src/test/java/org/apache/iceberg/TestSnapshotProducer.java

```diff
@@ -19,10 +19,17 @@
 package org.apache.iceberg;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+import java.io.IOException;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.util.PropertyUtil;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-public class TestSnapshotProducer {
+@ExtendWith(ParameterizedTestExtension.class)
```

Comment:
This extension is a custom one that I don't recommend using because it is a hack to bring forward parameterization behavior from JUnit4. Do we need to extend `TestBase`? Or should we just reference static fields from it?

Reply:
Seems fine to me. This is minor.